worlds/yoshisisland/Client.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import struct
+import typing
+import time
+from struct import pack
+
+from NetUtils import ClientStatus, color
+from worlds.AutoSNIClient import SNIClient
+
+if typing.TYPE_CHECKING:
+    from SNIClient import SNIContext
+else:
+    SNIContext = typing.Any
+
+snes_logger = logging.getLogger("SNES")
+
+GAME_YI = "Yoshi's Island"
+
+ROM_START = 0x000000
+WRAM_START = 0xF50000
+WRAM_SIZE = 0x20000
+SRAM_START = 0xE00000
+
+YOSHISISLAND_ROMHASH_START = 0x007FC0
+ROMHASH_SIZE = 0x0F
+
+ITEMQUEUE_HIGH = WRAM_START + 0x1465
+ITEM_RECEIVED = WRAM_START + 0x1467
+DEATH_RECEIVED = WRAM_START + 0x7E23B0
+GAME_MODE = WRAM_START + 0x0118
+YOSHI_STATE = SRAM_START + 0x00AC
+YI_DEATHLINK_ADDR = ROM_START + 0x06FC8C
+DEATHMUSIC_FLAG = WRAM_START + 0x004F
+DEATHFLAG = WRAM_START + 0x00DB
+DEATHLINKRECV = WRAM_START + 0x00E0
+GOALFLAG = WRAM_START + 0x14B6
+
+VALID_GAME_STATES = [0x0F, 0x10, 0x2C]
+
+class YISNIClient(SNIClient):
+    game = "Yoshi's Island"
+
+    async def deathlink_kill_player(self, ctx):
+        from SNIClient import DeathState, snes_buffered_write, snes_flush_writes, snes_read
+        game_state = await snes_read(ctx, GAME_MODE, 0x1)
+        if game_state[0] != 0x0F:
+            return
+
+        yoshi_state = await snes_read(ctx, YOSHI_STATE, 0x1)
+        if yoshi_state[0] != 0x00:
+            return
+
+
+        snes_buffered_write(ctx, WRAM_START + 0x026A, bytes([0x01]))
+        snes_buffered_write(ctx, WRAM_START + 0x00E0, bytes([0x01]))
+        await snes_flush_writes(ctx)
+        ctx.death_state = DeathState.dead
+        ctx.last_death_link = time.time()
+
+    async def validate_rom(self, ctx):
+        from SNIClient import snes_buffered_write, snes_flush_writes, snes_read

```

Comment:
```suggestion
        from SNIClient import snes_read

```
snes_read is the only SNIClient function used in validate_rom.